source/wp-content/themes/wporg-developer-blog/functions.php

```diff
@@ -41,3 +42,70 @@ function search_block_add_search_action( $block_content, $block ) {
 
 	return $block_content;
 }
+
+/**
+ * Outputs tags for the page head.
+ */
+function output_head_tags() {
+	$fields = [
+		// FYI: 'description' and 'og:description' are set further down.
+		'og:title'       => wp_get_document_title(),
+		'og:site_name'   => get_bloginfo( 'name' ),
+		'og:url'         => home_url( '/' ),
+		'twitter:title'  => wp_get_document_title(),
+		'twitter:site'   => '@WordPress',
+	];
+
+	$desc = '';
```

Comment:
Disregard if I'm misunderstanding, but can we not just use the excerpt?

Reply:
Thanks for pointing this out!

Originally this part was referred from a [snippet](https://github.com/WordPress/wporg-developer/blob/c186671beab3622f3c11b632bd3e281963616b82/source/wp-content/themes/wporg-developer-2023/inc/head.php#L111-L168) in wporg-developer. The reason it parsed `desc` and trimmed it down to <150 chars with regex from scratch is that [developer.w.org](http://developer.w.org/) doesn’t store data within the post_content in a way that get_the_excerpt would handle well. 

`get_the_excerpt()` would work well for the [is_singular() if branch](https://github.com/WordPress/wporg-developer/blob/c186671beab3622f3c11b632bd3e281963616b82/source/wp-content/themes/wporg-developer-2023/inc/head.php#L130), but [other parts](https://github.com/WordPress/wporg-developer/blob/c186671beab3622f3c11b632bd3e281963616b82/source/wp-content/themes/wporg-developer-2023/inc/head.php#L114-L129) it wouldn’t, so in the snippet the original author might just figure it was better to process all of the excerpts for SEO headers the same way.

I've changed to use [get_the_exceprt()](https://developer.wordpress.org/reference/functions/get_the_excerpt/) here as at the moment only `is_singular` is used and  I'd like to get this PR merged first to see if the original issue is fixed. 
Whether it needs some specific descriptions on different pages (front page, archive page, etc.), I'll have another discussion afterward to figure out the requirements.

Thanks to @dd32 for the clarification!